src/durmod/kaldi-durmod.cc

```diff
@@ -0,0 +1,463 @@
+                                                       (phone_duration - 1);
+  output_elements[0].push_back(std::make_pair(duration_id, 1.0));
+  SparseMatrix<BaseFloat> output_mat(output_dim, output_elements);
+
+  NnetIo input, output;
+  input.name = "input";
+  input.features = feat_mat;
+  input.indexes.resize(1);
+  output.name = "output";
+  output.features = output_mat;
+  output.indexes.resize(1);
+  eg->io.push_back(input);
+  eg->io.push_back(output);
+}
+
+void PhoneDurationEgsMaker::AlignmentToNnetExamples(
+                      const std::vector<std::pair<int32, int32> > &alignment,
+                      std::vector<NnetExample> *egs) const {
+// These lines are commented because we need edge examples too:
+//  if (alignment.size() < (model.left_ctx + model.right_ctx + 1)) {
+//    return;
+//  }
+  for (int i = 0; i < alignment.size(); i++) {
+    NnetExample eg;
+    MakeNnetExample(alignment, i, &eg);
+    // eg.io[0].indexes[0].t = i;
+    // eg.io[1].indexes[0].t = i;
+    egs->push_back(eg);
+  }
+}
+
+void PhoneDurationEgsMaker::InitFeatureMaker(const PhoneDurationModel &model) {
+  num_binary_features_ = model.questions_.size();
+  num_phone_identities_ = model.roots_.size() + 1;  // id=0 is for not-available
+                                                    // phones (i.e. null phones
+                                                    // which occur at edges).
+  left_ctx_ = model.left_context_;
+  right_ctx_ = model.right_context_;
+  max_duration_ = model.max_duration_;
+  int input_dim_phones = num_phone_identities_ * (left_ctx_ + right_ctx_ + 1);
+  int input_dim_durations = left_ctx_;
+  int input_dim_binary = num_binary_features_ * (left_ctx_ + right_ctx_ + 1);
+  feature_dim_ = input_dim_phones + input_dim_binary + input_dim_durations;
+
+  // create the reverse map for questions membership
+  for (int i = 0; i < model.questions_.size(); i++) {
+    for (int j = 0; j < model.questions_[i].size(); j++) {
+      int phone = model.questions_[i][j];
+      binary_feats_[phone].push_back(i);
+    }
+  }
+
+  // and the reverse map for phoneme tree roots
+  int max_phone_idx = 0;  // TODO(hhadian): in the end, check if all phones have
+                          // been categorized.
+  for (int i = 0; i < model.roots_.size(); i++) {
+    for (int j = 0; j < model.roots_[i].size(); j++) {
+      int phone = model.roots_[i][j];
+      KALDI_ASSERT(phone_id_[phone] == 0);  // only one root for each phone
+      phone_id_[phone] = i + 1;  // phone identities start from 1
+      if (max_phone_idx < phone)
+        max_phone_idx = phone;
+      if (binary_feats_.count(phone) <= 0) {
+        KALDI_WARN << "Phone " << phone << " does not have any "
+                   << "acoustic question associated.";
+      }
+    }
+  }
+}
+
+void PhoneDurationModel::InitNnet(int input_dim, int dim1,
+                                  int dim2, int output_dim) {
+  std::stringstream config;
+
```

Comment:
are you still using this code?  if so, maybe call dim1 and dim2 hidden_dim1 and hidden_dim2